packages/widget/src/components/network-selector/network-selector.ts

```diff
@@ -16,70 +17,91 @@ type Direction = (typeof Directions)[keyof typeof Directions];
 export class NetworkSelector extends LitElement {
   static styles = styles;
 
-  @property({
-    type: Boolean
-  })
-  disabled = false;
+  @state()
+  _isDropdownOpen = false;
 
-  @property({
-    type: Boolean
-  })
-  icons = true;
+  @property({ type: Boolean })
+  disabled = false;
 
-  @property({
-    type: String
-  })
+  @property({ type: String })
   direction?: Direction;
 
-  @property({
-    type: Object
-  })
-  selected?: Domain;
+  @property({ type: Object })
+  selectedNetwork?: Domain;
 
-  @property({
-    attribute: false
-  })
+  @property({ attribute: false })
   onNetworkSelected?: (network?: Domain) => void;
 
-  @property({
-    type: Array
-  })
+  @property({ type: Array })
   networks: Domain[] = [];
 
-  onChange({ target }: Event): void {
-    const { value } = target as HTMLOptionElement;
-    const network = this.networks.find((n) => String(n.chainId) == value);
-    this.onNetworkSelected?.(network);
+  _toggleDropdown = (): void => {
+    this._isDropdownOpen = !this._isDropdownOpen;
+  };
+
+  _selectOption(option: Domain, event: Event): void {
+    event.stopPropagation();
+    this.selectedNetwork = option;
+    this._isDropdownOpen = false;
+    this.onNetworkSelected?.(option);
   }
 
-  renderEntries(): Generator<unknown, void> | HTMLTemplateResult {
-    if (this.networks) {
-      return map(this.networks, (entry: Domain) => {
-        // TODO: render network icon
-        return html`<option value=${entry.chainId} class="network-option">
-          ${capitalize(entry.name)}
-        </option>`;
-      });
-    }
-    return html`<option selected value="">Network</option>`;
+  _renderNetworkIcon(name: string): HTMLTemplateResult {
+    return networkIconsMap[name] || networkIconsMap.default;
+  }
+
+  _renderEntries(): Generator<unknown, void> | HTMLTemplateResult {
+    return map(
+      this.networks,
+      (network: Domain) => html`
+        <div
+          class="dropdownOption"
+          @click="${(e: Event) => this._selectOption(network, e)}"
+          role="option"
+        >
+          ${this._renderNetworkIcon(network.name)}
+          <span class="networkName">${capitalize(network.name)}</span>
+        </div>
+      `
+    );
+  }
+
+  _renderTriggerContent(): HTMLTemplateResult {
+    return this.selectedNetwork
```

Comment:
here you can use `choose` or `when` directives

Reply:
Updated with "when"